src/sc/sidechainrpc.cpp

```diff
@@ -885,11 +885,13 @@ void ScRpcCmd::addInputs()
 
     static const bool fOnlyConfirmed = false;
     static const bool fIncludeZeroValue = false;
-    bool fProtectCoinbase = !Params().GetConsensus().fCoinbaseMustBeProtected;
-    static const bool fIncludeCoinBase = fProtectCoinbase;
-    static const bool fIncludeCommunityFund = fProtectCoinbase;
+    static bool fMustShieldCoinBase = ForkManager::getInstance().mustCoinBaseBeShielded(chainActive.Height());
```

Comment:
This variable now depends on runtime state, so it should not be `static` (and hence initialized only once).
```suggestion
    const bool fMustShieldCoinBase = ForkManager::getInstance().mustCoinBaseBeShielded(chainActive.Height());
```

Reply:
Totally agree, that was a bug. Addressed in ca71777097aece83e8f83375243baeac51c0451d.